ow_plexil/src/plans/FaultHandlingPattern3.plp

```diff
@@ -2,52 +2,60 @@
 // Research and Simulation can be found in README.md in the root directory of
 // this repository.
 
-// WARNING: This plan cannot work as intended until greater fault/action support
-// is added to the simulator. This serves mostly as a syntactic example.
-
-// DESIRED BEHAVIOR: If no faults are present, this plan will pan 180 degrees,
-// unstow and then stow the arm. If a fault is injected, the plan will pause 
-// until resolution.
-
-// LIMITATIONS: If an arm fault is triggered during Unstow, it will skip the Stow node. 
-
-// NOTE: If a fault is injected during an operation, that operation will only
-// halt immediately if the fault injected is associated with it directly,
-// i.e. an antenna fault during panning or an arm fault during
-// unstowing or stowing.
+// This plan illustrates the halting and resumption of a sequence in
+// the presence of faults.
 
+// If no faults are present this plan will, in sequence, pan the
+// antenna to 180 degrees, unstow the arm, and then stow it. If a
+// fault *relevant* to the current action is injected, the plan will
+// pause until the fault is cleared, and then continue the sequence.
+// Finally it prints the success status of each operation (i.e. did it
+// complete successfully).
 
 #include "ow-interface.h"
 
-FaultHandlingPattern3:
+FaultHandlingPattern3: UncheckedSequence
 {
+  Boolean PanSuccess;
+  Boolean UnstowSuccess;
+  Boolean StowSuccess;
 
   log_info ("Starting FaultHandlingPattern3 plan...");
 
   PanAntenna:
   {
     Start !Lookup(AntennaPanFault);
     Invariant !Lookup(AntennaPanFault);
+    Post Lookup(ActionGoalStatus("Pan")) == ACTION_SUCCEEDED;
 
     LibraryCall Pan (Degrees=180);
   }
 
+  PanSuccess = PanAntenna.outcome == SUCCESS;
+
   Unstow:
   {
     Start !Lookup(ArmFault);
```

Comment:
I believe this would make the plan wait for the antenna fault to be resolved before continuing, which sounds to be the intention from the description in the comment. 
```suggestion
    Start !Lookup(ArmFault) && !Lookup(AntennaPanFault);
```

Reply:
The comment has been revised.  Only the _relevant_ fault needs to be clear for each operation.